src/util.iced

```diff
@@ -0,0 +1,34 @@
+stream = require('stream')
+crypto = require('crypto')
+{make_esc} = require('iced-error')
+
+exports.stream_random_data = (strm, len, cb) ->
+  esc = make_esc(cb, "Error in stream writing")
+  written = 0
+  expected_results = []
+  while written < len
+    # generate random length
+    await crypto.randomBytes(1, esc(defer(index)))
+    amt = (index[0] + 1)*16
+
+    # generate random bytes of length amt
+    await crypto.randomBytes(amt, esc(defer(buf)))
```

Comment:
I think we talked about this before, but random input in tests makes test failures harder to reproduce. It would be one thing if the randomness was really pulling its weight, in practice our random inputs don't seem to be hitting the corner cases we need. The number of bugs in https://github.com/keybase/node-saltpack/pull/2 that the tests didn't catch still has me worried.